### PR TITLE
Added without_files flag to DeltaTable constructor

### PR DESCRIPTION
**Pull request by MykhailoHevak**
# Description
Added _without_files_ flag to DeltaTable constructor. If the flag is set it will be called method _without_files_ in the **DeltaTableBuilder**. And this allows users to control memory usage. Some append-only applications might have no need of tracking any files.

# Documentation
Indicates whether DeltaTable should not track files. This defaults to "false" (files are tracked). Some append-only applications might have no need of tracking any files. So, DeltaTable will be loaded with significant memory reduction.

### CODE REVIEW

File: python/deltalake/table.py [hunk 1]
```diff
@@ -89,7 +90,7 @@ def __init__(
         """
```
Comment (wjones127) [inline python/deltalake/table.py:90]: Will want to document this parameter:

```suggestion
        :param without_files: If True, will load table without tracking files. Some 
          append-only applications might have no need of tracking any files. So, the
          DeltaTable will be loaded with a significant memory reduction.
        """
```

File: python/deltalake/table.py [hunk 2]
```diff
@@ -77,6 +77,7 @@ def __init__(
         table_uri: str,
         version: Optional[int] = None,
         storage_options: Optional[Dict[str, str]] = None,
+        without_files: Optional[bool] = None
```
Comment (wjones127) [inline python/deltalake/table.py:80]: Why not make it clear what the default is in the signature?

```suggestion
        without_files: bool = False
```

**Review by wjones127 — CHANGES_REQUESTED** [review]
What's the behavior of `DeltaTable.file_uris()` when `without_files=True`? Does it return empty? Or raise an error? Or load the files? We might want to add a unit test for that in Python.

**Comment by wjones127** [comment]
> What's the behavior of DeltaTable.file_uris() when without_files=True? Does it return empty? Or raise an error? Or load the files? We might want to add a unit test for that in Python.

Just checked and it returns an empty list. We should document that for now, and consider changing that behavior in the future (#867).

**Comment by wjones127** [comment]
@MykhailoHevak looks like the merge you pushed overwrote the changes you accepted in the GitHub UI. Could you add those back?

**Comment by MykhailoHevak** [comment]
> @MykhailoHevak looks like the merge you pushed overwrote the changes you accepted in the GitHub UI. Could you add those back?

Oooops... I did it accidentally. Is it OK now?

**Comment by MykhailoHevak** [comment]
Do I have to do something else?

File: python/src/lib.rs [hunk 3]
```diff
@@ -106,6 +106,7 @@ impl RawDeltaTable {
         table_uri: &str,
         version: Option<deltalake::DeltaDataTypeLong>,
         storage_options: Option<HashMap<String, String>>,
+        without_files: bool
```
Comment (wjones127) [inline python/src/lib.rs:109]:
```suggestion
        without_files: bool,
```

**Review by wjones127 — APPROVED** [review]

**Comment by wjones127** [comment]
Thanks @MykhailoHevak!

**Comment by dennyglee** [comment]
Hey @MykhailoHevak I believe this is your first PR to the delta repo.  Can you ping me (dennyglee) via the [Delta Users slack](https://go.delta.io/slack) so we can send you some swag?  Thanks!

**Pull request merged**